src/ert/ensemble_evaluator/evaluator_tracker.py

```diff
@@ -144,7 +145,7 @@ def track(
                     indeterminate=self._model.isIndeterminate(),
                     progress=self._progress(),
                     iteration=iter_,
-                    snapshot=snapshot,
+                    snapshot=copy.deepcopy(snapshot),
```

Comment:
Why are we copying the snapshot here?

Reply:
that is a good question that we discussed the other day again as well. we try to explain a bit in the commit message:

    Make deep copy of full snapshot event
    
    This must be done as partial snapshots may modify the full snapshot held
    by the evaluator tracker.

basically a few lines above we have

```
    self._iter_snapshot[iter_] = snapshot
```

where we store the received snapshot as internal state in the evaluator tracker (a perhaps reasonable question at this point is why does the evaluator tracker need to store that state, too?) - when we yield it for further consumption, whatever grabs it can (and does) manipulate the snapshot (at least the metadata is manipulated by the gui).
that last point is also something that could / should be addressed in a refactor of the whole snapshot handling.

Reply:
Thanks for the nice explanation! When you are writting the issue for refactoring you could include "to consider removal of snapshot from evaluator tracker".

Reply:
included in [the issue](https://github.com/equinor/ert/issues/5866)